src/context.rs

```diff
@@ -161,7 +161,9 @@ impl Context {
 
         {
             let l = &mut *self.inner.scheduler.write().await;
-            l.start(self.clone()).await;
+            if let Err(err) = l.start(self.clone()).await {
+                error!(self, "Failed to start IO: {}", err)
```

Comment:
`start_io()` can only fail due to database error, there are no network operations. Actual connection will happen only when `prepare()` is called. If there is a database error and `start_io()` does not actually start the scheduler, it's better to report to the user I think, because it means messages will not be sent and received until the next `start_io()`. Normally this should not happen.